projects/plugins/jetpack/_inc/client/components/popover/index.jsx

```diff
@@ -79,12 +78,24 @@ class Popover extends Component {
 		bindWindowListeners();
 	}
 
+	isDOMNode( obj ) {
+		return obj && obj instanceof HTMLElement;
+	}
+
+	isRef( obj ) {
+		return obj && typeof obj === 'object' && 'current' in obj;
+	}
+
 	UNSAFE_componentWillReceiveProps( nextProps ) {
 		// update context (target) reference into a property
-		if ( ! isDOMElement( nextProps.context ) && nextProps.context !== null ) {
-			this.domContextRef.current = nextProps.context.current;
-		} else {
+
+		if ( this.isDOMNode( nextProps.context ) ) {
 			this.domContextRef.current = nextProps.context;
+		} else if ( this.isRef( nextProps.context ) ) {
+			this.domContextRef.current = nextProps.context.current;
+		} else if ( nextProps.context !== null ) {
+			// eslint-disable-next-line no-console
+			console.error( 'Expected a DOM node or a React ref', nextProps.context );
```

Comment:
Maybe we should mention "for props.context" or something like that?

(my bad for not including that in my earlier suggestion)

Reply:
Good point, added in cf0e911